src/solvers/dg.jl

```diff
@@ -732,50 +732,281 @@ end
+    @threaded for element in eachelement(dg, cache)
+        for j in eachnode(dg), i in eachnode(dg)
+            x_node = get_node_coords(cache.elements.node_coordinates, equations, dg,
+                                     i, j, element)
+            # For explanation of the implemented logic see comment above for the 1D case, i.e., 
+            # `compute_coefficients!` for `func::DiscontinuousFunction`, `mesh::AbstractMesh{1}`
+            # or the docstring of `DiscontinuousFunction`
+            if i == 1
+                x_node = SVector(nextfloat(x_node[1]), x_node[2])
+            elseif i == nnodes(dg)
+                x_node = SVector(prevfloat(x_node[1]), x_node[2])
+            end
+            if j == 1
+                x_node = SVector(x_node[1], nextfloat(x_node[2]))
+            elseif j == nnodes(dg)
+                x_node = SVector(x_node[1], prevfloat(x_node[2]))
+            end
+
+            u_node = func(x_node, t, equations)
+            set_node_vars!(u, u_node, equations, dg, i, j, element)
+        end
+    end
+end
+# Mesh-type general function for dragging the outer nodes inwards for 
+# discontinuous functions/initial conditions
+function compute_coefficients!(u, func::DiscontinuousFunction, t,
+                               mesh::AbstractMesh{2}, equations, dg::DG, cache)
     @threaded for element in eachelement(dg, cache)
+        # Helper variables to avoid quering `get_node_coords` too often
+        x1_min_at_node1 = true
+        x2_min_at_node1 = true
         for j in eachnode(dg), i in eachnode(dg)
-            x_node = get_node_coords(cache.elements.node_coordinates, equations, dg, i,
-                                     j, element)
+            x_node = get_node_coords(cache.elements.node_coordinates, equations, dg,
+                                     i, j, element)
+            # For explanation of the implemented logic see comment above for the 1D case, i.e., 
+            # `compute_coefficients!` for `func::DiscontinuousFunction`, `mesh::AbstractMesh{1}`
+            # or the docstring of `DiscontinuousFunction`
+            if i == 1
+                x_node_n = get_node_coords(cache.elements.node_coordinates, equations,
+                                           dg, nnodes(dg), j, element)
+
+                # For non-`TreeMesh`es the node orientation within an element is not necessarily
+                # consistent with the global coordinate system, thus we need to check what node
+                # lies on the "left" and which on the "right" side.
+                # NOTE: This still assumes that the first node index somewhat resembles the first coordinate direction!
+                if x_node[1] < x_node_n[1]
+                    x_node = SVector(nextfloat(x_node[1]), x_node[2])
+                else
+                    x1_min_at_node1 = false
+                    x_node = SVector(prevfloat(x_node[1]), x_node[2])
+                end
```

Comment:
This is probably the most convenient way of dragging the nodes inwards. As mentioned in the comment, this is probably not really fail-safe if a mesh is supplied/constructed where the `i` index does not really expand the element in `1` direction but `2` direction. I would assume that this is possible depending on the exact order of node numbering in a supplied mesh file. @andrewwinters5000 is that right?